### PR TITLE
Add hardhat scripts

### PR DESCRIPTION
**Pull request by DefiCake**

### CODE REVIEW

File: packages/solidity-contracts/scripts/deployAll.ts [hunk 1]
```diff
@@ -1,109 +0,0 @@
...
-
-// For localhost testing:
-//    - Spin up a node (http://127.0.0.1:8545/ by default):
-//        `npx hardhat node`
-//    - Run this script, pointing to localhost:
-//        `npx hardhat run --network localhost scripts/deploy.ts`
-
-// You can then connect to localhost (ethers, metamask, etc.) and the Fuel system will be deployed there at the addresses given
-
-const QUICK_DEPLOY = !!process.env.QUICK_DEPLOY;
```
Comment (PraneshASP) [inline packages/solidity-contracts/scripts/deployAll.ts:30]: why not just process.env.QUICK_DEPLOY?

File: packages/solidity-contracts/scripts/hardhat/depositETH.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { Wallet, parseEther } from 'ethers';
+import type { Signer } from 'ethers';
+import { isB256, isBech32, toB256 } from 'fuels';
+import { task } from 'hardhat/config';
+
+task('depositETH', 'grants a given role to a given adress')
```
Comment (PraneshASP) [inline packages/solidity-contracts/scripts/hardhat/depositETH.ts:6]: wrong description: 

`task('depositETH', 'grants a given role to a given adress')`

File: packages/solidity-contracts/scripts/hardhat/depositETH.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  .addFlag('env', 'use this flag to send transactions from env var PRIVATE_KEY')
```
Comment (PraneshASP) [inline packages/solidity-contracts/scripts/hardhat/depositETH.ts:7]: nit: can add a flag `i` (interactive) to enter private key on prompt for flexibility.

File: packages/solidity-contracts/scripts/hardhat/withdrawalBlacklist.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { Wallet, getBytes } from 'ethers';
+import type { Signer } from 'ethers';
+import { task } from 'hardhat/config';
+
+task('withdrawalBlacklist', 'pauses all l2 > l1 messages')
```
Comment (PraneshASP) [inline packages/solidity-contracts/scripts/hardhat/withdrawalBlacklist.ts:5]: wrong description.

File: packages/solidity-contracts/scripts/hardhat/withdrawalWhitelist.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { Wallet, getBytes } from 'ethers';
+import type { Signer } from 'ethers';
+import { task } from 'hardhat/config';
+
+task('withdrawalWhitelist', 'pauses all l2 > l1 messages')
```
Comment (PraneshASP) [inline packages/solidity-contracts/scripts/hardhat/withdrawalWhitelist.ts:5]: wrong description

**Review by PraneshASP — COMMENTED** [review]
few nits.

File: packages/solidity-contracts/scripts/hardhat/withdrawalBlacklist.ts (hunk 4 above)
Comment (DefiCake) [reply packages/solidity-contracts/scripts/hardhat/withdrawalBlacklist.ts]: Solved

File: packages/solidity-contracts/scripts/hardhat/depositETH.ts (hunk 2 above)
Comment (DefiCake) [reply packages/solidity-contracts/scripts/hardhat/depositETH.ts]: Solved

File: packages/solidity-contracts/scripts/hardhat/withdrawalBlacklist.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    const contract = await hre.ethers.getContractAt(
+      'FuelMessagePortalV3',
+      (
+        await hre.deployments.get('FuelMessagePortal')
+      ).address,
+      signer
+    );
+
+    const tx = await contract.addMessageToBlacklist(taskArgs.id);
```
[the remark this answers is not in the document]
Comment (DefiCake) [reply packages/solidity-contracts/scripts/hardhat/withdrawalBlacklist.ts:44]: Done

**Review by PraneshASP — APPROVED** [review]

**Pull request merged**